src/main/java/org/broadinstitute/hellbender/tools/exome/BayesianHetPulldownCalculator.java

```diff
@@ -0,0 +1,671 @@
+            } else {
+                baseCounts.put(base, 0);
+            }
+        }
+        return baseCounts;
+    }
+
+    /**
+     * Infer the alt allele base from the pileup. We choose the base with the highest frequency
+     * that is not Ref (this coincides with the maximum likelihood estimation).
+     *
+     * * Remark: while this is definitely not the perfect way to do it, it has no serious pitfalls:
+     *
+     * - if the input is balanced between betweet ref and alt, it correctly chooses alt.
+     * - if the input is all ref, then alt is chosen randomly, but it's OK since there are no alt counts
+     *   in the pileup anyway.
+     * - if the input is all alt, we are good.
+     * - if the input in all ref + a few errors, then alt will be choosen as the most frequent erroneous
+     *   base; again, this is OK because Hom likelihood is far greater than Het likelihood in this case
+     * - if the input is all alt + a few errors, we are good.
+
+     * @param baseQualities map from bases to error probabilities
+     * @param refBase the ref allele base
+     * @return the likely alt allele
+     */
+    @VisibleForTesting
+    public static Nucleotide inferAltFromPileup(final Map<Nucleotide, ArrayList<BaseQuality>> baseQualities,
+                                                final Nucleotide refBase) {
+
+        /* sort the bases in the descending order by their frequency */
+        Nucleotide[] bases = {Nucleotide.A, Nucleotide.C, Nucleotide.G, Nucleotide.T};
+        Arrays.sort(bases, (L, R) -> Integer.compare(baseQualities.get(R).size(), baseQualities.get(L).size()));
+        /* pick the base with highest frequency, skip over ref */
+        for (Nucleotide base : bases) {
+            if (base != refBase) {
+                return base;
+            }
+        }
+
+        /* we shouldn't be here unless the baseErrorProbabilities is malformed */
+        return Nucleotide.N;
+    }
+
+    /**
+     * For a normal or tumor sample, returns a data structure giving (intervals, reference counts, alternate counts),
+     * where intervals give positions of likely heterozygous SNP sites.
+     *
+     * The analysis for normal and tumor is identical. The IntervalList gives common SNP sites in 1-based
+     * format.
+     *
+     * The @hetCallingStrigency parameters sets the threshold Het posterior for calling:
+     *
+     *      hetPosteriorThreshold = 1 - 10^{-hetCallingStringency}
+     *      hetThresholdLogOdds = log(hetPosteriorThreshold/(1-hetPosteriorThreshold))
+     *                          = log(10^{hetCallingStringency} - 1)
+     *
+     * @param bamFile sorted BAM file for sample
+     * @param hetCallingStringency strigency for calling a Het site
+     * @return Pulldown of heterozygous SNP sites in 1-based format
+     */
+    @VisibleForTesting
+    public Pulldown getHetPulldown(final File bamFile, final double hetCallingStringency) {
```

Comment:
@mbabadi Note that very often you can at least make it `protected` instead of `public`.  Unlike C++ `protected` in Java means a method is exposed to subclasses *and* other classes in the same package, which almost always includes the test class.  Strictly speaking, you don't need any access modifier at all to give package classes access, and some people's preferred style is to omit the access modifier for `@VisibleForTesting` methods.  But whatever the case, it's rare that you ever have to make something `public` for testing.